src/components/grapher/GrapherTool.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <g class="grapher-tool--tool-dots-container">
+    <circle
+      v-for="dot in grapherToolDots"
+      :key="`${dot.x}-${dot.y}-tool--dot`"
+      class="grapher-tool--dot"
+      :cx="dot.x"
+      :cy="dot.y"
+      r="0.7"
+      data-test="grapher-tool--dot"
+    />
+    <g v-if="showDotLabels">
+      <text
+        v-for="(dot, index) in grapherToolDots"
+        :key="`${dot.x}-${dot.y}-tool--dottext`"
+        class="grapher-tool--dottext"
+        :x="dot.x"
+        :y="dot.y - 1"
+        data-test="grapher-tool--dottext"
+      >
+        {{ dotLabels[index] }}
+      </text>
+    </g>
+    <polyline
+      v-if="showSelectionLasso"
+      class="grapher-tool--selection-lasso"
+      :points="selectionLasso"
+      fill="none"
+      data-test="grapher-tool--selection-lasso"
+    />
+  </g>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import StuntSheetDot from "@/models/StuntSheetDot";
+
+/**
+ * Renders the field, the dots of the current stunt sheet, and pending dots
+ * generated from the tool in use
+ */
+export default Vue.extend({
+  name: "GrapherTool",
+  computed: {
+    grapherToolDots(): StuntSheetDot[] {
+      return this.$store.state.grapherToolDots;
+    },
+    showDotLabels(): boolean {
+      return this.$store.state.showDotLabels;
+    },
+    dotLabels(): string[] {
+      const dotLabels = this.$store.getters.getDotLabels;
+      const dots: StuntSheetDot[] = this.$store.state.grapherToolDots;
+      return dots.map((dot, index) => {
+        return dotLabels !== null &&
+          dot.dotLabelIndex !== null &&
+          dot.dotLabelIndex < dotLabels.length
+          ? dotLabels[dot.dotLabelIndex]
+          : index.toString();
```

Comment:
If no `dotLabelIndex` exists, then can you not show a label? An example where this is weird is if you select + move a dot that does not have a `dotLabelIndex`, or you try adding a dot and the tool dot has a label

Reply:
Yes!  sounds like a good idea.